ext/test/http/url_parser_test.cc

```diff
@@ -2,248 +2,136 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <gtest/gtest.h>
-#include <map>
 #include <string>
-#include <utility>
+#include <tuple>
 
 #include "opentelemetry/ext/http/common/url_parser.h"
 
 namespace http_common = opentelemetry::ext::http::common;
 
-inline const char *BoolToString(bool b)
+struct ParsedUrl
```

Comment:
I used a dedicated structure instead of `std::map<std::string, std::string>` because map's `operator[]` may change the map (if a key does not exist, it will create a default-constructed value and put it into the map). Test data must be immutable.